File: 1_GuidedExercise/idea_10.rb
class Idea10
  # Task: Use introduce parameter to replace the double and triple with a single function that takes an argument.
  def similar_functions(n)
    [double(n), triple(n)]
  end

  def double(n)
    n * 2
  end

  def triple(n)
    n * 3
  end
end

File: js/1_GuidedExercise/src/idea_10.js
// Task: Use introduce parameter to replace the double and triple with a single function that takes an argument.
const similar_functions = (n) => {
  return [double(n), triple(n)]
}

function double(n) {
  return n * 2
}

function triple(n) {
  return n * 3
}

module.exports = { similar_functions }
